installer/lib/phx_new.ex

```diff
@@ -46,123 +49,87 @@ defmodule Mix.Tasks.Phx.New do
-    # SQL spec, they are case insensitive unless quoted, which
-    # means creating a database like FoO is the same as foo in
-    # some storages.
-    {adapter_app, adapter_module, adapter_config} = get_ecto_adapter(db, String.downcase(app), app_mod)
-    pubsub_server = get_pubsub_server(app_mod)
-    in_umbrella? = in_umbrella?(proj_path, generator)
-    brunch_deps_prefix = if in_umbrella?, do: "../../../", else: "../"
-
-    adapter_config =
-      case Keyword.fetch(opts, :binary_id) do
-        {:ok, value} -> Keyword.put_new(adapter_config, :binary_id, value)
-        :error -> adapter_config
-      end
-    generator_config = generator_config(adapter_config)
-
-    binding = [app_name: app,
-               app_module: inspect(app_mod),
-               root_app_name: root_app,
-               root_app_module: inspect(root_mod),
-               web_app_name: web_app_name,
-               endpoint_module: inspect(Module.concat(web_namespace, Endpoint)),
-               web_namespace: inspect(web_namespace),
-               phoenix_dep: phoenix_dep(phoenix_path),
-               phoenix_path: phoenix_path,
-               phoenix_static_path: phoenix_static_path(phoenix_path),
-               pubsub_server: pubsub_server,
-               secret_key_base: random_string(64),
-               prod_secret_key_base: random_string(64),
-               signing_salt: random_string(8),
-               in_umbrella: in_umbrella?,
-               brunch_deps_prefix: brunch_deps_prefix,
-               brunch: brunch,
-               ecto: ecto,
-               html: html,
-               adapter_app: adapter_app,
-               adapter_module: adapter_module,
-               adapter_config: adapter_config,
-               hex?: Code.ensure_loaded?(Hex),
-               generator_config: generator_config,
-               namespaced?: namespaced?(app, app_mod)]
-
-    generator.gen_new(proj_path, binding)
-    copy_ecto(generator, app_path, binding)
-    copy_static(generator, web_path, binding)
-    copy_html(generator, web_path, binding)
-
+  defp prompt_to_install_deps(%Project{} = project) do
     install? = Mix.shell.yes?("\nFetch and install dependencies?")
+    starting_dir = File.cwd!()
 
-    File.cd!(proj_path, fn ->
+    maybe_cd(project.project_path, fn ->
       mix? = install_mix(install?)
-      File.cd!(web_path, fn ->
-        brunch? = install_brunch(install?)
-        extra   = if mix?, do: [], else: ["$ mix deps.get"]
-
-        print_mix_info(proj_path, extra)
+      extra = if mix?, do: [], else: ["$ mix deps.get"]
 
-        if binding[:ecto], do: print_ecto_info()
+      brunch? = maybe_cd(project.web_path, fn -> install_brunch(install?) end)
 
-        if not brunch?, do: print_brunch_info()
-      end)
+      print_mix_info(project.project_path, extra)
+      if Project.ecto?(project), do: print_ecto_info()
+      if not brunch?, do: print_brunch_info(project, starting_dir)
```

Comment:
I noticed this line is causing issues when doing:

```
cd apps && mix phx.new.ecto hello
```

```
** (ArgumentError) argument error
    lib/phx_new.ex:129: anonymous fn/3 in Mix.Tasks.Phx.New.prompt_to_install_deps/1
    (elixir) lib/file.ex:1162: File.cd!/2
    (mix) lib/mix/task.ex:294: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:58: Mix.CLI.run_task/2
    (elixir) lib/code.ex:370: Code.require_file/2
```

(looks like `brunch?` is then `nil` and so it fails on `if not nil, do:`)

I think we should use `if Project.brunch?(project)` here instead however it probably still needs to be modified as it returns `true` when doing `phx.new.ecto`